src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java

```diff
@@ -201,4 +202,21 @@ CaseNote updateOffenderCaseNote(@ApiParam(value = "Noms ID or Prisoner number (a
     PrivilegeSummary getLatestBookingIEPSummaryForOffender(
             @ApiParam(value = "offenderNo", required = true, example = "A1234AA") @PathVariable("offenderNo") @NotNull String offenderNo,
             @ApiParam(value = "Toggle to return IEP detail entries in response (or not).", required = true) @RequestParam(value = "withDetails", required = false, defaultValue = "false") boolean withDetails);
+
+    String NOMS_ID_REGEX_PATTERN = "[a-zA-Z][0-9]{4}[a-zA-Z]{2}";
+
+    @SuppressWarnings("RestParamTypeInspection")
+    @GetMapping("/{offenderNo}/transaction-history")
+    @ApiOperation(value = "Retrieve an offender's financial transaction history for cash, spends or savings.",
+            notes = "Transactions are returned in NOMIS order (Descending date followed by id).<br/>" +
+            "All transaction amounts are represented as pence values.") //todo order?, should use int for money?
```

Comment:
Did you mean to check this //todo in?

Reply:
I'm sorting by entryDate Descending and Sequence Ascending is that correct?

Reply:
Sounds right to me 👍